operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -42,9 +43,53 @@ func ValidateConfig(failOnMissingValidation bool, iopvalues map[string]interface
 	if err := validateFeatures(values, iopls).ToError(); err != nil {
 		warningMessage = fmt.Sprintf("feature validation warning: %v\n", err.Error())
 	}
+	warningMessage += deprecatedSettingsMessage(values)
 	return validationErrors, warningMessage
 }
 
+func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
+	global := values.GetGlobal()
+	pc := global.GetProxy()
+	messages := []string{}
+	addDeprecation := func(old, new string) {
+		messages = append(messages, fmt.Sprintf("! %s is deprecated; use %s", old, new))
+	}
+	if global.GetCertificates() != nil {
```

Comment:
You can use tpath.GetFromStructPath and a table to express this a bit more compactly.

Reply:
I agree in general but it doesn't handle ints/durations/strings that are not nil. I guess I can probably workaround that though, thanks for the suggestion. Can you take a look at the workaround (default values) to see if there is a better one?